File: problem-05/c/main.c
#include <stdio.h>
#include <stdbool.h>

int solution(int max_mod) {
	int output = max_mod;

	for (int i = max_mod; ; i += max_mod) {
		bool found = true;
		for (int j = max_mod; j > 1; j--) {
			if (i % j != 0) {
				found = false;
				break;
			}
		}

		if (found) {
			output = i;
			break;
		}
	}

	return output;
}

int main() {
	printf("Solution #5: %d\n", solution(20));
}

File: problem-05/rust/src/main.rs
fn main() {
    println!("Test: {}", solution(20));
}

fn solution(max_mod: i32) -> i32 {
    let mut output = max_mod;

    'outer: for x in (max_mod..).step_by(max_mod as usize) {
        for y in (1..=max_mod).rev() {
            if x % y != 0 {
                continue 'outer;
            }
        }

        output = x;
        break;
    }

    return output
}
